### PR TITLE
Encoding is ISO-8859-1

### LINKED ISSUE #12372: Source Salesforce: UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 8379: unexpected end of data
Is this your first time deploying Airbyte?: No
OS Version: debian-10
Memory / Disk: 16GB
Deployment: Docker
Airbyte Version: 0.36.1-alpha
Source name/version: Salesforce 1.0.3
Destination name/version: BigQuery 1.1.1
Step: Salesforce fails to sync
Description:
After running a Salesforce incremental sync for two weeks, everything worked fine until yesterday when I encountered the same error in product and staging. After retrying staging the error solved itself but this didn’t work in production.
I retried production more than 10 times already with different sync modes(incremental, full replication etc) but nothing worked. I also reset the incremental sync and started from scratch but that didn’t work as well. As a last resort, I deleted the Airbyte connection and all BigQuery data and tried running everything from scratch, but the error was still there.
Please note that production and staging are fetching from the same salesforce enviriment just writing to different BigQuery projects. Also, I have one Airbyte instance for both staging/prod.
See the logs below

[logs-529.txt](https://github.com/airbytehq/airbyte/files/8566408/logs-529.txt)

From Discourse: https://discuss.airbyte.io/t/salesforce-unicodedecodeerror/697

From user in Discourse:

> In [airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py line 276 1](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L276) I found the chunk ends with \xc3. Which is only part of the ü (\xc3\xbc) character which it should represent. So I think chunking and then decoding utf-8 causes the issue.

### PR DESCRIPTION
**Pull request by girarda**
## What
* We currently fail to decode some bytes received by salesforce. As mentioned by Marcos, downgrading down to 1.0.2 fixed the issue, which indicates it was introduced in https://github.com/airbytehq/airbyte/commit/644ace4036dee905179c2336318dc2f8450bccfa
* Fixes https://github.com/airbytehq/oncall/issues/221
* Fixes https://github.com/airbytehq/airbyte/issues/12372

## How
* Change the expected encoding from UTF-8 to ISO-8859-1 based on these 2 two threads which seem to indicate the encoding should be ISO-8859 
- https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/implementation_considerations.htm
- https://developer.salesforce.com/forums/?id=906F00000008lEyIAI
- https://developer.salesforce.com/forums/?id=906F000000090flIAA
* Add a unit test to confirm we can decode the byte we failed to decode in production

### CODE REVIEW

File: airbyte-integrations/connectors/source-salesforce/unit_tests/api_test.py [hunk 1]
```diff
@@ -525,3 +525,7 @@ def test_convert_to_standard_instance(stream_config, stream_api):
     bulk_stream = generate_stream("Account", stream_config, stream_api)
     rest_stream = bulk_stream.get_standard_instance()
     assert isinstance(rest_stream, IncrementalSalesforceStream)
+
+
+def test():
+    assert b"0\xe5".decode(SalesforceStream.encoding) == "0å"
```
Comment (girarda) [inline airbyte-integrations/connectors/source-salesforce/unit_tests/api_test.py:531]: confirmed this fails if `SalesforceStream.encoding` is 'utf-8'

**Review by brianjlai — APPROVED** [review]
➕ for the unit test for encoding

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
 class SalesforceStream(HttpStream, ABC):
     page_size = 2000
     transformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization)
+    encoding = "ISO-8859-1"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:37]: There is any link to Salesforce docs explaining or saying about this encoding?

**Review by marcosmarxm — APPROVED** [review]
thanks @girarda

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py (hunk 2 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:37]: https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/implementation_considerations.htm

**Comment by wissevrowl** [comment]
@girarda Different sources including your links indicate most Salesforce instances UTF-8 expcept for some instances

> If your org logs in to ssl.salesforce.com, your encoding is ISO-8859-1. All other instances use UTF-8. 

While your fix prevents the problem happening by switching to a non-variable length encoding, characters are now incorrectly decoded. 

For example I added '间单的说 🪐' in an opportunity's description. The Airbyte source now converts this to 'Ã©ÂÂ´Ã¥ÂÂÃ§ÂÂÃ¨Â¯Â´ Ã°ÂÂªÂ'

**Pull request merged**